src/map/utils/charutils.cpp

```diff
@@ -4470,15 +4471,30 @@ namespace charutils
             {
                 PChar->StatusEffectContainer->DelStatusEffect(EFFECT_DEDICATION);
             }
+        }
 
+        for (auto i = 2884; i <= 2892; ++i) // RHAPSODY KI are sequential, so start at WHITE and end at OCHRE
+        {
+            if (hasKeyItem(PChar, i))
+            {
+                bonus += 30;
```

Comment:
This bonus appears to be a static exp value being added to the base; instead of adding to the exp multiplier down below: